src/mapper_class.cpp

```diff
@@ -258,40 +235,30 @@ void MapperClass::Initialize(ros::NodeHandle *nh) {
     h_fade_thread_ = std::thread(&MapperClass::FadeTask, this);
     h_collision_check_thread_ = std::thread(&MapperClass::PathCollisionCheckTask, this);
     h_radius_collision_thread_ = std::thread(&MapperClass::RadiusCollisionCheck, this);
-    h_body_tf_thread_ = std::thread(&MapperClass::BodyTfTask, this, inertial_frame_id_, robot_frame_id_);
     // h_keyboard_thread_ = std::thread(&MapperClass::KeyboardTask, this);
 
-    // Subscriber for trajectories
+    // Subscribers -----------------------------------------------
     waypoints_sub_ = nh->subscribe<pensa_msgs::WaypointSet>
         ("/drone_arbiter/check_collision", 1, &MapperClass::WaypointsCallback, this);
     trajectory_status_sub_ = nh->subscribe<pensa_msgs::trapezoidal_p2pActionFeedback>
         ("/trapezoidal_p2p_action/feedback", 1, &MapperClass::TrajectoryStatusCallback, this);
+    base_link_pose_sub_ = nh->subscribe<geometry_msgs::PoseStamped>
+        (base_link_pose_topic, 5, &MapperClass::BaseLinkPoseCallback, this);
 
-    // Camera subscribers and tf threads ----------------------------------------------
-    for (uint i = 0; i < depth_cam_names.size(); i++) {
-        std::string cam_topic = depth_cam_prefix + depth_cam_names[i] + depth_cam_suffix;
-        cameras_sub_[i] = nh->subscribe<sensor_msgs::PointCloud2>
-              (cam_topic, 10, boost::bind(&MapperClass::CameraPclCallback, this, _1, i));
-        h_cameras_tf_thread_[i] = std::thread(&MapperClass::CameraTfTask, this, inertial_frame_id_, cam_frame_id[i], i);
-        ROS_INFO("[mapper] Subscribed to camera topic: %s", cameras_sub_[i].getTopic().c_str());
-    }
-
-    // Lidar subscribers and tf threads ----------------------------------------------
-    for (uint i = 0; i < lidar_names.size(); i++) {
-        std::string lidar_topic = lidar_prefix + lidar_names[i] + lidar_suffix;
-        lidar_sub_[i] = nh->subscribe<sensor_msgs::PointCloud2>
-              (lidar_topic, 10, boost::bind(&MapperClass::LidarPclCallback, this, _1, i));
-        h_lidar_tf_thread_[i] = std::thread(&MapperClass::LidarTfTask, this, inertial_frame_id_, lidar_frame_id[i], i);
-        ROS_INFO("[mapper] Subscribed to lidar topic: %s", lidar_sub_[i].getTopic().c_str());
-    }
+    // Lidar synchronized subscriber
+    lidar_sync_sub_ = new message_filters::Subscriber<sensor_msgs::PointCloud2>(*nh, lidar_topic, 10);
+    base_link_pose_sync_sub_ =
+        new message_filters::Subscriber<geometry_msgs::PoseStamped>(*nh, base_link_pose_topic, 10);
```

Comment:
Do we want to use a queue for these topics? We may want to always use the very latest data for syncing. Can the sync policy have a queue, even if these subs don't? Not quite sure how these two queue sizes interact.

Reply:
I didn't pay much attention to this when I wrote the code, but after reading the [documentation](http://wiki.ros.org/message_filters/ApproximateTime) it seems like it would suffice to have the both with a `queue` of size 2 to best synchronize them, do you also understand it this way? A queue of 10 would just be slightly less efficient, but it wouldn't make a different